src/schematic/traits/autoscaler.rs

```diff
@@ -201,6 +202,10 @@ impl TraitImplementation for Autoscaler {
             match client.request::<hpa::HorizontalPodAutoscaler>(req) {
                 Ok(hpa) => hpa,
                 Err(e) => {
+                    if e.to_string().contains("NotFound") {
```

Comment:
Please try not to use magic string literals. The content of this error message is actually critical.

Reply:
This is not any magic string literals. This is the fixed field "reason" returned as a string by the kube::Error when any queried object is not found. Similar check have been done in multiple places such as instigator.rs (Line: 697, 735) before this fix. Though I agree it would be better to use error code here instead of reason string.

Reply:
A logic based on a reason string is brittle, there is no guarantee that the server won't tweak  the string wording in the future that can break us. Couple with the fact that there is not a single test to validate and safeguard future change makes it even less desirable.  This is probably okey for a reference implementation.

Reply:
The kube::Error provides the guarantee that "NotFound" string will be returned at reason field. If that is changed, it will break not just our implementation but anyone relying on Kube::Error guarantees. However, we will make change to be not relying on a string in the next PR. Also, feel free to add any missing test gap you find.